File: samples/syntax/sample.py
"""A small Python sample."""
from dataclasses import dataclass


@dataclass
class Point:
    x: int
    y: int

    def norm2(self) -> int:
        return self.x * self.x + self.y * self.y


if __name__ == "__main__":
    print(Point(3, 4).norm2())

File: samples/syntax/sample.rs
struct Point {
    x: i32,
    y: i32,
}

impl Point {
    fn norm2(&self) -> i32 {
        self.x * self.x + self.y * self.y
    }
}

fn main() {
    println!("{}", Point { x: 3, y: 4 }.norm2());
}
